src/js/components/chat-input.jsx

```diff
@@ -38,9 +38,17 @@ export class ChatInputComponent extends Component {
 
     resizeInput(attempt = 0) {
         const node = findDOMNode(this);
-        if (node.offsetWidth - this.refs.button.offsetWidth > 0) {
+
+        const nodeRect = node.getBoundingClientRect();
+        const buttonRect = this.refs.button.getBoundingClientRect();
+
+        // use floor on widget and ceil on button to get worst rounding scenario
```

Comment:
I see...
The comment is a bit confusing.
Maybe something like... // floor widget width and ceil button width to ensure button width <= widget width
Why can't they be decimals, though?